bpfd-operator/controllers/bpfd-agent/xdp-program_test.go

```diff
@@ -424,43 +198,153 @@ func TestXdpProgramControllerCreateMultiIntf(t *testing.T) {
+		// owningConfig Label was correctly set
+		require.Equal(t, bpfProgEth1.Labels[internal.BpfProgramOwnerLabel], name)
+		// node Label was correctly set
+		require.Equal(t, bpfProgEth1.Labels[internal.K8sHostLabel], fakeNode.Name)
+		// Finalizer is written
+		require.Equal(t, r.getFinalizer(), bpfProgEth1.Finalizers[0])
+		// Type is set
+		require.Equal(t, r.getRecType(), bpfProgEth1.Spec.Type)
+		// Require no requeue
+		require.False(t, res.Requeue)
+
+		// Update UID of bpfProgram with Fake UID since the fake API server won't
+		bpfProgEth1.UID = types.UID(fakeUID1)
+		err = cl.Update(ctx, bpfProgEth1)
+		require.NoError(t, err)
+
+		// Fifth reconcile should create the second bpfProgram.
+		res, err = r.Reconcile(ctx, req)
+		if err != nil {
+			t.Fatalf("reconcile: (%v)", err)
+		}
+
+		// Require no requeue
+		require.False(t, res.Requeue)
+
+		// Sixth reconcile should update the second bpfProgram's status.
+		res, err = r.Reconcile(ctx, req)
+		if err != nil {
+			t.Fatalf("reconcile: (%v)", err)
+		}
+
+		// Require no requeue
+		require.False(t, res.Requeue)
+
+		uuid1 := string(bpfProgEth1.UID)
+
+		expectedLoadReq1 := &gobpfd.LoadRequest{
+			Bytecode: &gobpfd.BytecodeLocation{
+				Location: &gobpfd.BytecodeLocation_File{File: bytecodePath},
+			},
+			Name:        sectionName,
+			ProgramType: *internal.Xdp.Uint32(),
+			Metadata:    map[string]string{internal.UuidMetadataKey: uuid1, internal.ProgramNameKey: name},
+			MapOwnerId:  nil,
+			Attach: &gobpfd.AttachInfo{
+				Info: &gobpfd.AttachInfo_XdpAttachInfo{
+					XdpAttachInfo: &gobpfd.XDPAttachInfo{
+						Priority:  0,
+						Iface:     fakeInts[1],
+						ProceedOn: []int32{2, 31},
+					},
+				},
+			},
+		}
+
+		// Check that the bpfProgram's maps was correctly updated
+		err = cl.Get(ctx, types.NamespacedName{Name: bpfProgName1, Namespace: metav1.NamespaceAll}, bpfProgEth1)
+		require.NoError(t, err)
+
+		// prog ID should already have been set
+		id1, err := bpfdagentinternal.GetID(bpfProgEth1)
+		require.NoError(t, err)
+
+		// Check the bpfLoadRequest was correctly built
+		if !cmp.Equal(expectedLoadReq1, cli.LoadRequests[int(*id1)], protocmp.Transform()) {
+			t.Logf("Diff %v", cmp.Diff(expectedLoadReq1, cli.LoadRequests[int(*id1)], protocmp.Transform()))
+			t.Fatal("Built bpfd LoadRequest does not match expected")
+		}
+
+		require.Nil(t, bpfProgEth0.Spec.Maps)
+
+		// Check that the bpfProgram's maps was correctly updated
+		err = cl.Get(ctx, types.NamespacedName{Name: bpfProgName1, Namespace: metav1.NamespaceAll}, bpfProgEth1)
+		require.NoError(t, err)
+
+		require.Nil(t, bpfProgEth1.Spec.Maps)
+
+		// Check that the bpfProgram's status was correctly updated
+		err = cl.Get(ctx, types.NamespacedName{Name: bpfProgName1, Namespace: metav1.NamespaceAll}, bpfProgEth1)
+		require.NoError(t, err)
+
+		require.Equal(t, string(bpfdiov1alpha1.BpfProgCondLoaded), bpfProgEth1.Status.Conditions[0].Type)
+	}
+}
+
+func TestXdpProgramControllerCreate(t *testing.T) {
```

Comment:
Don't we need to make these same changes for the tc program unit tests as well?

Reply:
We could, but I wasn't changing the tc cases, so I didn't do it.